Add details of Examine customization breaking change in Umbraco 16 (#7534)
umbraco/UmbracoDocs: 17/umbraco-cms/fundamentals/setup/upgrading/version-specific/README.md

Umbraco 16 adds support for asynchronous migrations and part of this work involved creating a new base class for package migrations. This leads to a source-compatible but binary-incompatible breaking change. In practice, this means that package code using migrations and calling base class helper methods such as `TableExists` can be recompiled without change. But if built against 15 and run on 16, a "method missing" exception will be thrown. For more details on the feature and the changes implemented, see the [PR](https://github.com/umbraco/Umbraco-CMS/pull/17057).

**Examine is now registered via a composer**

> [A new abstraction and implementation for search](https://github.com/umbraco/Umbraco.Cms.Search) is being worked on in an external package. To support this, we have implemented a means of disabling the default Examine based search in Umbraco. This has required moving the Examine component registration to a composer.

Reviewer: ```suggestion
[A new abstraction and implementation for search](https://github.com/umbraco/Umbraco.Cms.Search) is being worked on in an external package. To support this, a method has been implemented to disable the default Examine-based search in Umbraco. This has required moving the Examine component registration to a composer.
```
